front/app/components/CardBlock.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+class CardBlock extends React.Component {
+
+    render() {
+
+        var stackElements = [];
+
+        if (this.props.stacks) {
+
+            const stacks = this.props.stacks.split(/[\s,;-]+/).map(
+                element => element.trim()
+            );
+
+            stacks.forEach(stack => {
+
+                const stackName = stack.trim();
+
+                if (stackName) {
+                    stack = require(`../../public/images/${stackName}.png`);
+                    stackElements.push(
+                        <img className="img-circle" src={stack}/>
+                    );
+                }
+            });
+        }
+
+        return (
+            <div className="card-block stacks">
+                {stackElements}
```

Comment:
globalement tu peux simplifier la construction du stackElements comme ça : 
const stacksElements = this.props.stacks.split(...).map(le trim).filter(element => element).map(element => (<img className="..." src="la source" />));

Voila, le filter c'est pour virer les potentiel nom vide après le trim. 
Globalement les transformations de ce genre on essaye de les faire en mode programmation fonctionnelle (map, reduce, filter, ...) c'est plus concis, lisible et ça évite les effets de bords parce qu'on ne change jamais directement ce qui nous est passé en entrée.